core/evm/sputnik/sputnik.go

```diff
@@ -152,16 +151,14 @@ Loop:
 					adapter.updateStorage(changedAcc.Address(), key, value)
 				}
 			}
-			adapter.setCode(changedAcc.Address(), changedAcc.Code())
+			adapter.setAccount(changedAcc.Address(), changedAcc.Balance().Uint64(), changedAcc.Code(), changedAcc.Nonce().Uint64())
```

Comment:
This code can be efficient more. You have an instance of the account before: `createContractAccount` you need too just update it....